### ISSUE TITLE
FeignClientsRegistrar fail with latest spring

### ISSUE THREAD

**Issue by igreenfield** [spring-cloud/spring-cloud-netflix#1352]
```java
private void validate(Map<String, Object> attributes) {
		if (StringUtils.hasText((String) attributes.get("value"))) {
			Assert.isTrue(!StringUtils.hasText((String) attributes.get("name")),
					"Either name or value can be specified, but not both");
			Assert.isTrue(!StringUtils.hasText((String) attributes.get("serviceId")),
					"Either name (serviceId) or value can be specified, but not both");
		}
		if (StringUtils.hasText((String) attributes.get("name"))) {
			Assert.isTrue(!StringUtils.hasText((String) attributes.get("serviceId")),
					"Either name or serviceId can be specified, but not both");
		}
	}
```
this method throw: java.lang.IllegalArgumentException: Either name or value can be specified, but not both

while my source look:

```scala
@FeignClient("player")
trait PlayerFeign {
  @RequestMapping(value = Array("playerId", "count"), method = Array(RequestMethod.POST), produces = Array("application/json"))
  def getAllArmsFrom(@RequestParam playerId: String, @RequestParam count: Int)
}
```

**Comment by spencergibb** [comment]
What version of the Spring Cloud Release Train are you using and what is the version of spring?

**Comment by igreenfield** [comment]
```xml
<parent>
        <groupId>org.springframework.boot</groupId>
        <artifactId>spring-boot-starter-parent</artifactId>
        <version>1.4.0.RELEASE</version>
        <relativePath/> <!-- lookup parent from repository -->
</parent>

<dependency>
            <groupId>org.springframework.cloud</groupId>
            <artifactId>spring-cloud-starter-feign</artifactId>
            <version>1.1.6.RELEASE</version>
</dependency>
```

spring 4.3.2.RELEASE

**Comment by ryanjbaxter** [comment]
You should be configuring your POM as follows (this is for Brixton.SR5)
```
<parent>
    <groupId>org.springframework.boot</groupId>
    <artifactId>spring-boot-starter-parent</artifactId>
    <version>1.4.0.RELEASE</version>
</parent>
<dependencyManagement>
    <dependencies>
        <dependency>
            <groupId>org.springframework.cloud</groupId>
            <artifactId>spring-cloud-dependencies</artifactId>
            <version>Brixton.SR5</version>
            <type>pom</type>
            <scope>import</scope>
        </dependency>
    </dependencies>
</dependencyManagement>
<dependencies>
  <dependency>
    <groupId>org.springframework.cloud</groupId>
    <artifactId>spring-cloud-starter-feign</artifactId>
  </dependency>
</dependencies>
```

**Comment by spencergibb** [comment]
Fixed in master in afb43c5883d4c7e486c5dd4e868afa368dc65007 and in 1.1.x via 75dcd309e9213c361312240f9726ee053444a905
